_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp

```diff
@@ -653,10 +653,15 @@ namespace ExtBuffer
 
             return false;
         }
-        if (buf.PicWidthInLumaSamples == 0)
-            buf.PicWidthInLumaSamples  = align2_value(par.mfx.FrameInfo.CropW > 0 ? (mfxU16)(par.mfx.FrameInfo.CropW + par.mfx.FrameInfo.CropX)  : par.mfx.FrameInfo.Width,  codedPicAlignment);
 
-        if (buf.PicHeightInLumaSamples == 0)
+        bool needUpdate = false;
+        if (buf.PicWidthInLumaSamples > par.mfx.FrameInfo.Width || buf.PicHeightInLumaSamples > par.mfx.FrameInfo.Height)
```

Comment:
Ok, so user requested size bigger than in initialization, right? If so, we should return with error, or it turns out that we silently ignore buffer passed by user... Or why should we update ```PicWidthInLumaSamples``` but not ```mfx.FrameInfo.Width```. Still unclear for me

Reply:
mfx.FrameInfo.Width is updated correctly when rotated, while PicWidthInLumaSamples has initial value and we get an error in CheckVideoParam(). Having the initial resolution of the stream 1920x1080, we get 1088x1080 after this function instead of 1088x1920, because the correct value of par.mfx.FrameInfo.Height is compared to the incorrect value of PicHeightInLumaSamples, and then par.mfx.FrameInfo.Height equals PicHeightInLumaSamples.

Reply:
well, if so I believe you need to update values at the same time when ```mfx.FrameInfo.Width```. Where in code this adjustment happens?

Reply:
This happens in the app. You're right, it's better to update it there. I'll close this pull request.